cpp/include/raft_runtime/neighbors/ivf_pq.hpp

```diff
@@ -20,15 +20,24 @@
 
 namespace raft::runtime::neighbors::ivf_pq {
 
-#define RAFT_INST_SEARCH(T, IdxT)                            \
-  void search(raft::device_resources const&,                 \
-              const raft::neighbors::ivf_pq::search_params&, \
-              const raft::neighbors::ivf_pq::index<IdxT>&,   \
-              const T*,                                      \
-              uint32_t,                                      \
-              uint32_t,                                      \
-              IdxT*,                                         \
-              float*,                                        \
+#define RAFT_INST_SEARCH(T, IdxT)                                        \
+  void search(raft::device_resources const&,                             \
```

Comment:
Please remove the pointer-based API from `raft::runtime`. We only need the mdspan API now.